File: mentorship/recommendation/week#2/snow.py
n = int(input())

piles = list(map(int, input().split()))
heat = list(map(int, input().split()))


for i in range(n):
	ans = 0
	for j in range(i+1):
		if piles[j]>0:
			melted = piles[j] - heat[i]
			if(melted < 0): melted = piles[j]
			else: melted = heat[i]
			ans+= melted
			piles[j] -= heat[i]
		if(piles[j]<0): piles[j] = 0
	print(ans, end=" ")
print()
	

File: mentorship/recommendation/week#2/snow.cpp
#include <bits/stdc++.h>
using namespace std;
int n;
int piles[100005], heat[100005];

int main(){
	scanf("%d", &n);
	
	for (int i = 0; i < n; ++i)
		scanf("%d", &piles[i]);

	for (int i = 0; i < n; ++i)
		scanf("%d", &heat[i]);


	for (int i = 0; i < n; ++i){
		int ans = 0;
		for (int j = 0; j < i+1; ++j){
			if(piles[j] > 0){
				int melted = piles[j] - heat[i];
				if(melted < 0) melted = piles[j];
				else melted = heat[i];

				ans += melted;
				piles[j] -= heat[i];
			}

			if(piles[j] < 0) piles[j] = 0;
		}

		cout<<ans<<" ";
	}
	cout<<"\n";
	return 0;
}